android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -633,27 +694,26 @@ public WritableMap stopProfiling() {
         final WritableMap result = new WritableNativeMap();
         File output = null;
         try {
+            AndroidProfiler.ProfileEndData end = androidProfiler.endAndCollect(false, null);
             HermesSamplingProfiler.disable();
 
             output = File.createTempFile(
                 "sampling-profiler-trace", ".cpuprofile", reactApplicationContext.getCacheDir());
-
             if (isDebug) {
                 logger.log(SentryLevel.INFO, "Profile saved to: " + output.getAbsolutePath());
             }
 
-            try (final BufferedReader br = new BufferedReader(new FileReader(output));) {
-                HermesSamplingProfiler.dumpSampledTraceToFile(output.getPath());
+            HermesSamplingProfiler.dumpSampledTraceToFile(output.getPath());
+            result.putString("profile", readStringFromFile(output));
 
-                final StringBuilder text = new StringBuilder();
-                String line;
-                while ((line = br.readLine()) != null) {
-                    text.append(line);
-                    text.append('\n');
-                }
+            WritableMap androidProfile = new WritableNativeMap();
+            byte[] androidProfileBytes = FileUtils.readBytesFromFile(end.traceFile.getPath(), maxTraceFileSize);
```

Comment:
is this happening in the main thread (i'm not really into react)?
In the Android layer we do it when building the envelope to send

Reply:
Currently, the RN Native Modules are not running on the main thread. But the RN docs recommends avoiding that assumption and spawning a worker thread.

But since this is a sync function also the JS code using this is sync, we would anyway have to block the thread to wait for the worker. Unless we make the JS code async too. Does that sound plausible? @stefanosiano

Reply:
Example profile: https://sentry-sdks.sentry.io/profiling/profile/sentry-react-native/0cd899c326af461b9d20a73408eaee81/flamegraph/?colorCoding=by+system+vs+application+frame&fov=0%2C0%2C5139650048%2C41&query=&sorting=call+order&tid=23160&view=top+down

Reply:
We can keep it as is for now. We can reevaluate if native modules will run on the main thread, to avoid other complexities (profile is still required to be sent in the same envelope of the transaction).
Also, continuous profiling (or a custom Android profiler) will likely change this, so let’s keep it as is